src/backend/main.ts

```diff
@@ -102,6 +102,38 @@ export async function connect(config: connections.Config) {
   toast.success('Connected to forwarder successfully!')
 }
 
+// Adam Chen snapshot 1.0 helper functions
+
+function numberToBytes(number: number) {
+  // you can use constant number of bytes by using 8 or 4
+  const len = Math.ceil(Math.log2(number) / 8)
+  const byteArray = new Uint8Array(len)
+  for (let index = 0; index < byteArray.length; index++) {
+    const byte = number & 0xff
+    byteArray[index] = byte
+    number = (number - byte) / 256
+  }
+  return byteArray
+}
+function bytesToNumber(byteArray: Uint8Array) {
+  let result = 0
+  for (let i = byteArray.length - 1; i >= 0; i--) {
+    result = result * 256 + byteArray[i]
+  }
+
+  return result
+}
+
+/**
+ * Returns timestamp in seconds
+ */
+function getTimestamp() {
+  // get time in seconds
+  return Math.floor(Date.now() / 1000)
```

Comment:
We can keep milliseconds if no specific reason for seconds, because most NDN modules use milliseconds (except a small part of ndn-cert). Life will be easier if we keep everywhere the same.

Reply:
Understood, will use miliseconds.